pages/ekmann.js

```diff
@@ -0,0 +1,352 @@
+          {page === 9 && question[9]}
+        </p>
+
+        {page === 0 && (
+          <StyledTextarea
+            type="text"
+            name="0"
+            value={formData["0"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 1 && (
+          <StyledTextarea
+            type="text"
+            name="1"
+            value={formData["1"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 2 && (
+          <StyledTextarea
+            type="text"
+            name="2"
+            value={formData["2"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 3 && (
+          <StyledTextarea
+            type="text"
+            name="3"
+            value={formData["3"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 4 && (
+          <StyledTextarea
+            type="text"
+            name="4"
+            value={formData["4"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 5 && (
+          <StyledTextarea
+            type="text"
+            name="5"
+            value={formData["5"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 6 && (
+          <StyledTextarea
+            type="text"
+            name="6"
+            value={formData["6"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 7 && (
+          <StyledTextarea
+            type="text"
+            name="7"
+            value={formData["7"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 8 && (
+          <StyledTextarea
+            type="text"
+            name="8"
+            value={formData["8"]}
+            onChange={handleChange}
+            placeholder="write here..."
+          />
+        )}
+        {page === 9 && (
+          <>
+            <StyledTextarea
+              type="text"
+              name="9"
+              value={formData["9"]}
+              onChange={handleChange}
+              placeholder="write here..."
+              required
+            />
+            <p>{emptyFieldsError}</p>
+          </>
+        )}
+
+        {page !== 9 && (
+          <NavButton
+            handleClick={() => {
+              setPage((currPage) => currPage + 1);
+            }}
+          >
+            next
+          </NavButton>
+        )}
+        {page === 9 && (
+          <div>
+            <NavButton handleClick={handleSubmit}>Analyze Emotion</NavButton>
+
+            <TbList onClick={() => setShowList(!showList)} />
+            {showList && (
```

Comment:
I think you can avoid this state by making the rendering of the emotionResult depending on if it's not an empty string. if there is content, you tender the results, if there is none, you don't :)

Reply:
The showList is not what you think it is. You have a Circles Visualisation and a results list. This showList opens the result list, which is normally hidden.